File: frontend/src/app/project-inverter.ts
import { Inverter } from "./inverter";

export class ProjectInverter {
    id?:number;
    serialNumber:String;
    project:{
        id:number;
    };
    inverter:{
        id:number;
    };
     
    constructor(serialnumber:String="",pid:number=1,iid:number=1){
        this.serialNumber=serialnumber;
        this.project={id:pid};
        this.inverter={id:iid};
    }
    
}

File: solarbackend/src/main/java/solar/solarbackend/entity/ProjectInverter.java
package solar.solarbackend.entity;



import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;

@Entity
@Data
@Table(name = "Project_Inverter")
public class ProjectInverter {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "inveter_id")
    private Long id;

    @Column(name = "serial_number", unique = true)
    private String serialNumber;

    @JsonBackReference(value = "project-id")
    @ManyToOne
    @JoinColumn(name = "project_id")
    private Project project;

    @JsonBackReference(value = "inverter")
    @ManyToOne
    @JoinColumn(name = "inverter" )
    private Inverters inverter;
}
